ARCHITECTURE.md

```diff
@@ -0,0 +1,73 @@
+# Exonum Architecture
+
+This document describes the high-level architecture of Exonum. Its target
+audience is the developers of Exonums core. To learn how to *use* Exonum, see
+https://exonum.com/doc/.
+
+This document is provided on the best effort bases -- it is not guranteed to be
+up-to-date and complete. The purpose of this document is to highlight the most
+important types, modules and functions in the code, to allow new contributors to
+dive into Exonum quickly. It is assumed that the reader is familiar with core
+concepts of Exonum, such as services and transactions.
+
+
+## Code organization
+
+The repository is a single Cargo workspace of several related packages:
+
+- `exonum`, most of the code lives here,
+
+- `sandbox`, where integration tests live,
+
+- `testkit`, a framework for testing Exonum services.
+
+
+This document currently describes only the `exonum` crate.
+
+
+## Storage
+
+Storage module can be found at `src/storage`. It provides an abstracton over an
+embeded key value store: the `Database` trait. The keys and values of are just
+slices of bytes: `&[u8]`.
+
+To transform raw bytes into native Rust datastructures, the `StorageKey` and
```

Comment:
Won't it be better to use active voice here to explicitly say _who_ uses these traits? The core doesn't need them, but a storage user -- does.

Reply:
@dmitry-timofeev: core itself uses storage to store the blockchain itself, so it does make use of StorageKey/Value. 

I'll try to rephrase this a bit and give more concrete pointers to usage sites inside the Core.